.github/workflows/benchmark-local.yaml

```diff
@@ -0,0 +1,73 @@
+# cli command to run this from repo
+# given a run, get output (link?)
+# input variable to CLI to allow users to specify where to output their artefacts
+
+name: Run performance comparisons
+
+on:
+  workflow_dispatch:
+    inputs:
+      c1:
+        description: First commit hash
+        required: true
+      c2:
+        description: Second commit hash
+        required: true
+
+jobs:
+  build:
+    runs-on: [self-hosted, linux, x64, bench]
+    strategy:
+      fail-fast: true
+      matrix:
+        commit: ["${{ github.event.inputs.c1 }}", "${{ github.event.inputs.c2 }}"]
+    timeout-minutes: 15
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+    - name: Assume AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: us-west-2
+        role-to-assume: ${{ secrets.ACTIONS_AWS_ROLE_ARN }}
+        role-session-name: daft-performance-comparisons
```

Comment:
Why is this necessary -- Is this for S3 upload permissions?

Reply:
Yes, this was here for AWS S3 uploads. I believe writes to the bucket are private (whereas reads are public).